piecemeta/piecemeta.go

```diff
@@ -127,23 +126,31 @@ func (ps *PieceMeta) addIndexForPiece(ctx context.Context, pieceCid cid.Cid, dea
 	// Get a reader over the piece data
 	reader, err := ps.sealer.GetReader(ctx, dealInfo.SectorID, dealInfo.PieceOffset, dealInfo.PieceLength)
 	if err != nil {
-		return err
+		return fmt.Errorf("getting reader over piece %s: %w", pieceCid, err)
 	}
 
-	// Get an index from the CAR file - works for both CARv1 and CARv2
-	idx, err := car.ReadOrGenerateIndex(reader, car.ZeroLengthSectionAsEOF(true), car.StoreIdentityCIDs(true))
+	// Iterate over all the blocks in the piece to extract the index records
+	recs := make([]model.Record, 0)
+	opts := []carv2.Option{car.ZeroLengthSectionAsEOF(true), car.StoreIdentityCIDs(true)}
```

Comment:
we're reading, so i dont' think the `StoreIdentityCIDs` option does anything here